modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js

```diff
@@ -97,17 +99,47 @@ export default function DatePicker({
 	 * Updates the rawDate state whenever the prop value or localizedValue changes,
 	 * but it keep user's input case theres no language change.
 	 */
-	useEffect(
-		() =>
-			setDate(({formattedDate, name, predefinedValue, rawDate}) =>
-				name === date.name &&
-				predefinedValue === date.predefinedValue &&
-				rawDate === ''
-					? {...date, formattedDate}
-					: date
-			),
-		[date]
-	);
+	useEffect(() => {
+		const handleClickEvent = () => {
+			const isValidMomentFormat = moment(
+				formattedDate,
+				momentFormat,
+				true
+			).isValid();
+			const saveButton = document.querySelector('.btn-primary');
+			const datePickerDropdown = document.querySelector(
+				'.date-picker-dropdown-menu.show'
+			);
```

Comment:
We usually don't get these elements directly from the DOM.

Can you think of different solutions to avoid getting the elements directly from the DOM?

see:  https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#validating_forms_using_javascript